chef/cookbooks/provisioner/recipes/base.rb

```diff
@@ -301,6 +301,21 @@
         %x{zypper --non-interactive addrepo --refresh #{attrs[:url]} #{name}}
       end
     end
+    # make sure the chosen additional packages are available in the repos
+    target_platform_name = "suse-" + node["platform_version"].sub(".", "_")
+    packages = node[:provisioner][:suse][:autoyast][:packages][target_platform_name]
+    packages.each do |package|
+      package_exists = false
+      repos.each do |name, repo|
+        if system("zypper search -r #{repo[:url]} --match-exact -t package #{package}")
+          package_exists = true
+          break
+        end
+      end
+      unless package_exists
+        raise "Missing additional package #{package} on #{node[:fqdn]}. Abort!"
+      end
+    end
```

Comment:
So I'm confused by this block: this will be executed on nodes (not the admin server); which more or less means that the node will already have been installed and that the autoyast bit will already have been used. So this comes too late.

Or am I missing the purpose of the block?